### ISSUE TITLE
CertificateBuilder::validity() flexibility

### ISSUE THREAD

**Issue by shutton** [Devolutions/picky-rs#75]
This method takes only a `picky::x509::date::UTCDate`, which might be a problem due to [RFC5280 Section 4.1.2.5](https://tools.ietf.org/html/rfc5280#section-4.1.2.5), which specifies that dates before 2050 be encoded as a UTCTime, and >= 2050 be encoded as a GeneralizedTime.

I ran into this when trying connect to a server with a custom certificate using the built-in `curl` on macOS, which reported this error:
```
curl: (60) SSL certificate problem: format error in certificate's notBefore field
More details here: https://curl.haxx.se/docs/sslcerts.html
```
I think that's actually a bug in that version of curl, but it brought the issue to my attention, as I tried to build a certificate using a `picky_asn1::date::UTCTime` and found I couldn't.

**Comment by bcortier-devolutions** [comment]
Good catch! Looks like I missed this part.

**Comment by bcortier-devolutions** [comment]
Addressed by 7bbc60c806264dc0c2150 and publish in `picky-6.1.1`
No code change is required, `picky_asn1::date::UTCTime` will be encoded as `UTCTime` or `GeneralizedTime` for `Validity` depending on the date year
Let me know if you find anything else! :slightly_smiling_face:
